src/cascadia/WindowsTerminal/IslandWindow.cpp

```diff
@@ -884,23 +884,36 @@ void IslandWindow::_RestoreFullscreenPosition(const RECT rcWork)
                rcWork.left - _rcWorkBeforeFullscreen.left,
                rcWork.top - _rcWorkBeforeFullscreen.top);
 
+    const til::size ncSize{ GetTotalNonClientExclusiveSize(dpiWindow) };
+
+    RECT rcWorkAdjusted = rcWork;
+
+    // GH#10199 - adjust the size of the "work" rect by the size of our borders.
+    // We want to make sure the window is restored within the bounds of the
+    // monitor we're on, but it's totally fine if the invisible borders are
+    // outside the monitor.
+    rcWorkAdjusted.left -= ncSize.width<long>() / 2;

```

Comment:
The optimizer will probably catch it... but I'm not a huge fan of doing the division twice...

Reply:
meh, okay

Reply:
I mean whatever. I'm not your mom.